### PR TITLE
Add FIPS doc

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/DMTF/libspdm/issues/2219

### CODE REVIEW

**Comment by steven-bellock** [comment]
Feel free to merge as you and @xiaoyuruan see fit.

File: doc/fips.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# FIPS support
```
Comment (xiaoyuruan) [inline doc/fips.md:1]: FIPS -> FIPS 140-3

File: doc/fips.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+libspdm integrator is expected to choose crypto module and support [Cryptographic Module Validation Program (CMVP)](https://csrc.nist.gov/Projects/cryptographic-module-validation-program).
+
+## Design
+
+### Cryptographic Algorithm Self-Test
+
+| SPDM Algorithm      | Specification                             | Type | Test Attributes      |
+| ------------------- | ----------------------------------------- | ---- | -------------------- |
+| SHA-256/384/512     | [NIST.FIPS.180-4](https://doi.org/10.6028/NIST.FIPS.180-4)   | KAT  | SHA-256/512          |
```
Comment (xiaoyuruan) [inline doc/fips.md:15]: Expand KAT (Known Answer Test) at first use.

File: doc/fips.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+| SHA3-256/384/512    | [NIST.FIPS.202](https://doi.org/10.6028/NIST.FIPS.202)     | KAT  | SHA3-256             |
+| RSA-SSA             | [NIST.FIPS.186-5](https://doi.org/10.6028/NIST.FIPS.186-5), [rfc8017](https://tools.ietf.org/html/rfc8017)   | KAT  | RSA-SSA2048 + SHA256 |
+| RSA-PSS             | [NIST.FIPS.186-5](https://doi.org/10.6028/NIST.FIPS.186-5), [rfc8017](https://tools.ietf.org/html/rfc8017)   | PCT  | RSA-PSS2048 + SHA256 |
```
Comment (xiaoyuruan) [inline doc/fips.md:18]: Expand PCT (Pairwise Consistency Test) at first use.

File: doc/fips.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
+# FIPS support
+
+libspdm 3.0.0 starts adding [FIPS 140-3](https://csrc.nist.gov/publications/detail/fips/140/3/final) support for [Cryptographic Algorithm Validation Program (CAVP)](https://csrc.nist.gov/Projects/cryptographic-algorithm-validation-program). The FIPS boundary is libspdm library plus crypto library.
```
Comment (xiaoyuruan) [inline doc/fips.md:3]: CAVP is an "offline" test program - NIST provides vectors and vendor provides answers for NIST to check.  This can be done as part of validation, and no specific implementation required.

KAT and PCT and mode output and version output are all for CMVP.

Suggest removing "for CAVP".

Comment (xiaoyuruan) [inline doc/fips.md:3]: >The FIPS boundary is libspdm library plus crypto library.

If this is true, then libspdm should implement integrity selftest and automatically run libspdm_fips_run_selftest() at load (dllmain).

File: doc/fips.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+ * `libspdm_fips_selftest_ecdsa()`
+ * `libspdm_fips_selftest_eddsa()`
+
+If any test failed, then `libspdm_fips_run_selftest()` will return false.
+
+`libspdm_fips_run_selftest()` requires `fips_selftest_context` parameter, which is initialized by `libspdm_get_fips_selftest_context_size()`, `libspdm_init_fips_selftest_context()` in [spdm_common_lib](https://github.com/DMTF/libspdm/blob/main/include/library/spdm_common_lib.h).
+
+The expected step is as follows:
+1) The integrator invokes `libspdm_get_fips_selftest_context_size()` and `libspdm_init_fips_selftest_context()` to create the FIPS selftest context.
+2) The integrator invokes `libspdm_fips_run_selftest()` to trigger self-test.
```
Comment (xiaoyuruan) [inline doc/fips.md:114]: As commented above, libspdm_fips_run_selftest() should be triggered by libspdm itself.  Is the following sample code part of libspdm?

File: doc/fips.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
+# FIPS support
+
+libspdm 3.0.0 starts adding [FIPS 140-3](https://csrc.nist.gov/publications/detail/fips/140/3/final) support for [Cryptographic Algorithm Validation Program (CAVP)](https://csrc.nist.gov/Projects/cryptographic-algorithm-validation-program). The FIPS boundary is libspdm library plus crypto library.
+
+NOTE: libspdm did not get any FIPS certificate.
```
Comment (xiaoyuruan) [inline doc/fips.md:5]: suggest rewording:
The SPDM WG does not plan to obtain FIPS 140-3 CMVP or CAVP certification for libspdm.

File: doc/fips.md (hunk 1 above)
Comment (jyao1) [reply doc/fips.md:1]: Done

File: doc/fips.md (hunk 4 above)
Comment (jyao1) [reply doc/fips.md:3]: done

Comment (jyao1) [reply doc/fips.md:3]: OK. Then we can remove this.

Currently, I dont think we support DLL.

File: doc/fips.md (hunk 5 above)
Comment (jyao1) [reply doc/fips.md:114]: No.
We added it before, but it seems cause lots of confusing. Finally, we decide to let integrator invoke libspdm_fips_run_selftest() in 3.0.0.

We can revisit later if there is need.

File: doc/fips.md (hunk 6 above)
Comment (jyao1) [reply doc/fips.md:5]: done.

**Comment by jyao1** [comment]
@xiaoyuruan , I have updated according to your feedback.
Please review it again.

**Review by xiaoyuruan — APPROVED** [review]
